### PR TITLE
Add GetRandomValue to Cache

### PR DESCRIPTION
**Pull request by muXxer**
Hi coocood,
I need to get some random entrys from the cache for a project.

Feel free to merge it if you like the solution.

Regards, muXxer

### CODE REVIEW

**Comment by coocood** [comment]
@muXxer 
Thank you for the PR.
But the `Random ` function is not thread-safe.

**Comment by muXxer** [comment]
@coocood 
Sorry for that!
Should be fixed now.

**Comment by kxlt** [comment]
Any updates on this?

File: segment.go [hunk 1]
```diff
@@ -14,6 +16,15 @@ var ErrLargeKey = errors.New("The key is larger than 65535")
 var ErrLargeEntry = errors.New("The entry size is larger than 1/1024 of cache size")
 var ErrNotFound = errors.New("Entry not found")
 
+var randLock = &sync.Mutex{}
+var rng = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+func Random(min, max int) int {
```
Comment (coocood) [inline segment.go]: I think we don't need to export this function.

**Comment by coocood** [comment]
@muXxer 
There is a conflict.

**Comment by muXxer** [comment]
@coocood 
Solved the conflicts and updated to your latest codebase.
Sorry for the delay. I totally forgot this PR.

**Comment by coocood** [comment]
@muXxer 
You can try this https://github.com/coocood/rtutil/blob/master/rtutil.go#L32
It's thread-safe and no lock.

**Comment by coocood** [comment]
And we need to cover the functions in unit test.

**Comment by bernardoaraujor** [comment]
Hi @coocood  and @muXxer !

I am trying to build https://gitlab.com/semkodev/hercules/
which depends on @muXxer's freecache repo (that seems to no longer exist).

I am facing this error:
```
| src/gitlab.com/semkodev/hercules/tangle/tip.go:174:27: TipsCache.GetRandomValue undefined (type *freecache.Cache has no field or method GetRandomValue)
```
seems to be related to this PR.

**Comment by muXxer** [comment]
@bernardoaraujor
You should post this issue in the Hercules repository. Unfortunately it is abandoned, but there will be an alternative coming soon. Heracles would not be compatible with IRI1.8

**Comment by bernardoaraujor** [comment]
Oh I didn't know that, thanks for letting me know!
Do you know what the alternative will be?

**Comment by muXxer** [comment]
Maybe better join the official discord rather than using this GitHub repository for a chat ;)

**Pull request closed without merging**